clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/bitbucket/BitbucketArtifactAccount.java

```diff
@@ -17,14 +17,31 @@
 
 package com.netflix.spinnaker.clouddriver.artifacts.bitbucket;
 
+import com.google.common.base.Strings;
 import com.netflix.spinnaker.clouddriver.artifacts.config.ArtifactAccount;
 import com.netflix.spinnaker.clouddriver.artifacts.config.BasicAuth;
-import lombok.Data;
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
+import javax.annotation.ParametersAreNullableByDefault;
+import lombok.Builder;
+import lombok.Value;
+import org.springframework.boot.context.properties.ConstructorBinding;
 
-@Data
+@NonnullByDefault
+@Value
 final class BitbucketArtifactAccount implements ArtifactAccount, BasicAuth {
-  private String name;
-  private String username;
-  private String password;
-  private String usernamePasswordFile;
+  private final String name;
+  private final String username;
+  private final String password;
+  private final String usernamePasswordFile;
+
+  @Builder
+  @ConstructorBinding
+  @ParametersAreNullableByDefault
+  BitbucketArtifactAccount(
+      String name, String username, String password, String usernamePasswordFile) {
+    this.name = Strings.nullToEmpty(name);
```

Comment:
is it valid to have an artifact account with no name? (generally asking this question to all the artifact account implementations). It is probably worth a not-null-or-empty check where we can add legitimate constraint checking

Reply:
I actually don't think that we prevent a null name anywhere at this point, so would be a bit wary of asserting that as part of this PR.  That also goes for adding more assertions to these constructors for other fields that can't be null...while in general I think we probably should do these assertion checks at the time of object creation, we are currently failing downstream at the time of use.  So moving constraint checks here would have the effect of breaking startup for people who have misconfigured accounts but weren't using them.  I guess we could also catch and warn on failed account creation to mitigate that...but this is getting out of the scope of what I really wanted to bite of with this PR.

Reply:
Fair enough